### ISSUE TITLE
CreateAndPopulate incorrectly does not recognize clearml in a requirements.txt if supplied with extras

### ISSUE THREAD

**Issue by cajewsa** [allegroai/clearml#867]
## Describe the bug
`CreateAndPopulate` incorrectly does not recognize clearml in a requirements.txt if supplied with extras. This leads to clearml overriding my "clearml" dependency, and thus throwing away version requirements.

## To reproduce
Make a requirements.txt with e.g. `clearml[azure]==1.7.*` and create a Task using `Task.create`. When creating the task, the "installed packages" specify a new `clearml` added to the list:
```
clearml[azure]==1.7.*
clearml
```
which will end up taking precedence, and ignoring the version specifier.

## Expected behaviour
I expect no modifications of my requirements if clearml is specified, even with extras.
The error lies within `clearml.backend_interface.task.populate.CreateAndPopulate.create_task` (line 245) ([link](https://github.com/allegroai/clearml/blob/master/clearml/backend_interface/task/populate.py#L259))
where a fix is likely either
- `if package.startswith("clearml")`, or
- `package = reduce(lambda a, b: a.split(b)[0], "#;@=~<>", line).strip()` adding `[` and `]` as splitters
- matching the extras pattern (`clearml[*]`) more explicitly with a regex.

## Environment
* Server type: self hosted
* ClearML SDK: 1.7.2
* ClearML Server Version: 1.7.0-232
* Python Version: 3.10.6
* OS: Linux

**Comment by cajewsa** [comment]
Temporary workaround is to add my Azure extras separately instead:
```
clearml==1.7.*
azure-storage-blob
```
but this beats the purpose of having the extras option :-)

**Comment by ainoam** [comment]
Thanks for catching @cajewsa.

Care to do the honors with a PR?

**Comment by pollfly** [comment]
Hey @cajewsa! Just letting you know that this issue has been resolved in the recently released [v1.9.1](https://github.com/allegroai/clearml/releases/tag/v1.9.1). Let us know if there are any issues :)
